test/e2e/framework/framework.go

```diff
@@ -179,7 +174,6 @@ func (f *Framework) BeforeEach() {
 
 	// The fact that we need this feels like a bug in ginkgo.
 	// https://github.com/onsi/ginkgo/v2/issues/222
```

Comment:
nit, the comment no longer applies

Reply:
Right. This begs the question: should I keep the if check at all?

The only situation where it is not nil is in the situation that you brought up recently elsewhere:
```
f := framework.NewDefaultFramework
f.ClientSet = <some client set>
```

If there's exactly one `It` after that, then it will use this client set provided by the setup code. But if there's a second `It`, then the framework.AfterEach will clear it and framework.BeforeEach will set it up anew.

Also, in a normal Ginko suite, where is this `<some client set>` supposed to come from? The code above runs at init time, before command line parsing. That makes it impossible to configure how to connect to a cluster.

Overall I think we should not support such code and simply remove the if check.

Reply:
I have pushed a version that drops the obsolete comment and the if check.

Reply:
yeah, I don't quite understand the logic it seems it is 
1. start test - create clientset
2. run test
3. end test, nil clientset

In theory, I think we should create the clientset once we start and reuse it ... that is BeforeSuite IIUIC 
https://onsi.github.io/ginkgo/#suite-setup-and-cleanup-beforesuite-and-aftersuite

anyway, I think your reasoning is correct and this was this way before :shrug:

Reply:
The clientset might be using different options for each test, therefore it has be created anew:
https://github.com/kubernetes/kubernetes/blob/2b2be7fa6b69c7591a31475fc4fb1dcff362a128/test/e2e/framework/framework.go#L204-L205

I also prefer it that way because then tests are better separated from each other.